serverless/networking/serverless-ossm-tls.adoc

```diff
@@ -0,0 +1,180 @@
++
+
+[source,terminal]
+----
+$ openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj '/O=Example Inc./CN=example.com' -keyout example.com.key -out example.com.crt
+----
+
+. Create a certificate signing request for your domain:
++
+
+[source,terminal]
+----
+$ openssl req -out custom.example.com.csr -newkey rsa:2048 -nodes -keyout custom.example.com.key -subj "/CN=custom-ksvc-domain.example.com/O=Example Inc."
+----
+
+. Sign the request with your CA:
++
+
+[source,terminal]
+----
+$ openssl x509 -req -days 365 -CA example.com.crt -CAkey example.com.key -set_serial 0 -in custom.example.com.csr -out custom.example.com.crt
+----
+
+. Check that the certificates appear in your directory:
++
+
+[source,terminal]
+----
+$ ls -1
+----
+
++
+
+.Example output
+[source,terminal]
+----
+custom.example.com.crt
+custom.example.com.csr
+custom.example.com.key
+example.com.crt
+example.com.key
+----
+
+. Create a secret:
++
+
+[source,terminal]
+----
+$  oc create -n istio-system secret tls custom.example.com --key=custom.example.com.key --cert=custom.example.com.crt
+----
+
+. Attach the secret to the Istio ingress gateway, by editing the `ServiceMeshControlPlane` resource.
+.. Edit the `ServiceMeshControlPlane` resource:
++
+
+[source,terminal]
+----
+$ oc edit -n istio-system ServiceMeshControlPlane <control_plane_name>
+----
+
+.. Check that the following lines exist in the resource, and if they do not, add them:
++
+
+[source,yaml]
+----
+istio-ingressgateway:
+  secretVolumes:
+  - mountPath: /custom.example.com
+    name: custom-example-com
+    secretName: custom.example.com
+----
+
+. Update the Istio ingress gateway to use your secret.
+.. Edit the `default-gateway` resource:
++
+
+[source,terminal]
+----
+$ oc edit gateway default-gateway
+----
+
+.. Check that the following lines exist in the resource, and if they do not, add them:
++
+
+[source,yaml]
+----
+- hosts:
+  - custom-ksvc-domain.example.com
+  port:
+    name: https
+    number: 443
+    protocol: HTTPS
+  tls:
+    mode: SIMPLE
+    privateKey: /custom.example.com/tls.key
+    serverCertificate: /custom.example.com/tls.crt
+----
+
+. Update the route to use pass-through TLS and `8443` as the `spec.port.targetPort`.
+.. Edit the route:
++
+
+[source,terminal]
+----
+$ oc edit route -n istio-system hello
+----
+
+. Add the following configuration to the route:
++
+
+[source,yaml]
+----
+spec:
+  host: custom-ksvc-domain.example.com
+  port:
+    targetPort: 8443
+  tls:
+    insecureEdgeTerminationPolicy: None
+    termination: passthrough
+  to:
+    kind: Service
+    name: istio-ingressgateway
+    weight: 100
+  wildcardPolicy: None
+----
+
+.Verification steps
+
+. Access your serverless application by a secure connection that is now trusted by the CA:
++
+
+[source,terminal]
+----
+$ curl --cacert example.com.crt -H "Host: custom-ksvc-domain.example.com" --resolve "custom-ksvc-domain.example.com:443:<ip_address>" https://custom-ksvc-domain.example.com:443
```

Comment:
I propose to change this to this command:
```suggestion
$ curl --cacert example.com.crt \
  --connect-to "custom-ksvc-domain.example.com:443:${INGRESS_ROUTER}:443" \
  https://custom-ksvc-domain.example.com
```

It's shorter, and works with hostname and IP. Previous one required IP only.

I needed to call:

```
curl --cacert example.com.crt \
  -H "Host: custom-ksvc-domain.example.com" \
  --resolve "custom-ksvc-domain.example.com:443:$(getent hosts ${INGRESS_ROUTER} | awk '{ print $1 }' | head -n 1)" \
  https://custom-ksvc-domain.example.com
```

Reply:
Guess I missed the comment, idk. Added it now, but is `${INGRESS_ROUTER}` part of the command? If this is a variable like `<ip_address>` previously it needs to be in the same format for OCP docs, part of the guidelines. I.e. `<ingress_router>`

Reply:
It is an Ingress router IP or hostaname.